tests/test_config.py

```diff
@@ -1868,6 +1868,46 @@ def test_quick_search_model_specific(self):
         with self.assertRaises(TritonModelAnalyzerException):
             self._evaluate_config(args, yaml_content, subcommand='profile')
 
+    def test_model_weightings(self):
+        """
+        Test that weightings can be specified only per model in the YAML
+        """
+        args = [
+            'model-analzyer', 'profile', '--model-repository', 'cli-repository',
```

Comment:
your test uses profile. Are we confident it works if the user specifies the analyze step instead?

Reply:
No. I've added unit tests to protect this.